### ISSUE TITLE
Unstable geometry type for buffer operation

### ISSUE THREAD

**Issue by bathyspheric** [Toblerity/Shapely#584]
I understand that the buffer operation must change some geometries, for example Point -> Polygon,
but is it expected behaviour to change MultiPolygon -> Polygon?

## Expected behavior and actual behavior.

I expected a buffer of a MultiPolygon of length 1 to return a MutliPolygon object of length 1.
Instead `buffer` returned a Polygon object.

## Steps to reproduce the problem.

```python 
mp = MultiPolygon([Point(0, 0).buffer(1)])
new_mp = mp.buffer(1)
assert mp.type == new_mp.type, "{} != {}".format(mp.type, new_mp.type)
# AssertionError: MultiPolygon != Polygon
```

## Operating system
 Mac OS X 10.13.4

## Shapely version and provenance
1.6.4.post1 from conda-forge

**Comment by sgillies** [comment]
@bathyspheric yes, this is the behavior that we inherit from GEOS. The algorithms of that library return single part geometries when possible. I can see how this could be unexpected.

While it would be possible to make multi- type preservation an option for `buffer` and other methods, the issue might be more directly solved by writing a utility function that takes geometry objects and returns their multi-part equivalents. Then we could do 

```
new_mp = multi(mp.buffer(1))
assert new_mp.type == 'MultiPolygon'  # True
```
